libraries/state_history/include/eosio/state_history/log_catalog.hpp

```diff
@@ -93,6 +93,11 @@ class log_catalog {
                      "block ${b} is before first block ${s} of ${name}.log",
                      ("b", block_num)("s", retained_log_files.begin()->begin_block_num)("name", retained_log_files.begin()->path_and_basename.string()));
 
+         //check if this log already has the same blockid at the given blocknum. This is indicative of a resync or replay and there is no need to write the
+         // same log entry again. otherwise we risk unrotating and blowing away existing log files
+         if(get_block_id(block_num) == id)
+            return;
```

Comment:
I don't think we need to add this feature now, but I have had to provide a patch in the past to bypass this behavior so that a corrupted SHiP log could be repaired by replaying from a snapshot before the corrupted log.

Reply:
yep that reminds me of AntelopeIO/leap#2393

Probably we should add something to `spring-util` to allow simple mutations of the log similar to the block log.